File: evidence/generated-systems/golang/manufacturing-on-golang/manufacturing-on-golang/internal/model/BOMItem.go
package model

import (
    "gorm.io/gorm"
)

//==============================================================
// BOMItem Declaration
//==============================================================
type BOMItem struct {
    gorm.Model
     LineNumber                                                            string
    Quantity                                                            string
    ScrapPercent                                                            string
    BomId         *uint
    Bom           *BOM `gorm:"foreignKey:BomId"`
    ComponentId         *uint
    Component           *Item `gorm:"foreignKey:ComponentId"`

// parent associations as their child

}

File: evidence/generated-systems/django/manufacturing-on-django/manufacturingOnDjango/models/BOMItem.py
from django.db import models

#======================================================================
# 
# Encapsulates data for model BOMItem
#
# @author Harbormaster Dev Team
#
#======================================================================

#======================================================================
# Class BOMItem Declaration
#======================================================================
class BOMItem (models.Model):

#======================================================================
# attribute declarations
#======================================================================
	lineNumber = models.IntegerField(null=True)
	quantity = Quantity
	scrapPercent = Percentage
	bom = models.ForeignKey('BOM', on_delete=models.CASCADE, null=True, blank=True, related_name='+')
	component = models.ForeignKey('Item', on_delete=models.CASCADE, null=True, blank=True, related_name='+')

#======================================================================
# function declarations
#======================================================================
	def toString(self):
		str = ""
		str = str + self.lineNumber
		str = str + self.quantity
		str = str + self.scrapPercent
		return str;
    
	def __str__(self):
		return self.toString();

	def identity(self):
		return "BOMItem";
    
	def objectType(self):
		return "BOMItem";
